pkg/agent/openflow/fields.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import binding "antrea.io/antrea/pkg/ovs/openflow"
+
+// Fields using reg.
+var (
+	// reg0 (NXM_NX_REG0)
+	// reg0[0..3]: Field to mark the packet source. Marks in this field include,
+	//   - 0: from the tunnel port
+	//   - 1: from antrea-gw0
+	//   - 2: from the local Pods
+	//   - 4: from the Bridge interface
+	//   - 5: from the uplink interface
+	PktSourceField     = binding.NewRegField(0, 0, 3, "PacketSource")
+	FromTunnelRegMark  = binding.NewRegMark(PktSourceField, 0)
+	FromGatewayRegMark = binding.NewRegMark(PktSourceField, 1)
+	FromLocalRegMark   = binding.NewRegMark(PktSourceField, 2)
+	FromUplinkRegMark  = binding.NewRegMark(PktSourceField, 4)
+	FromBridgeRegMark  = binding.NewRegMark(PktSourceField, 5)
+	// reg0[16]: Mark to indicate the ofPort number of an interface is found.
+	OFPortFoundRegMark = binding.NewOneBitRegMark(0, 16, "OFPortFound")
+	// reg0[17]: Mark to indicate the packet needs to be SNATed with Node's IP.
+	SNATNodeIPRegMark = binding.NewOneBitRegMark(0, 17, "SNATWithNodeIP")
+	// reg0[18]: Mark to indicate the packet needs DNAT to virtual IP.
+	// If a packet uses HairpinRegMark, it will be output to the port where it enters OVS pipeline in L2ForwardingOutTable.
+	HairpinRegMark = binding.NewOneBitRegMark(0, 18, "Hairpin")
+	// reg0[19]: Mark to indicate the packet's MAC address needs to be rewritten.
+	RewriteMACRegMark = binding.NewOneBitRegMark(0, 19, "RewriteMAC")
+	// reg0[20]: Mark to indicate the packet is denied(Drop/Reject).
+	CnpDenyRegMark = binding.NewOneBitRegMark(0, 20, "CNPDeny")
+	// reg0[21..22]: Field to indicate disposition of Antrea Policy. It could have more bits to support more disposition
+	// that Antrea policy support in the future.
+	// Marks in this field include,
+	//   - 0b00: allow
+	//   - 0b01: drop
+	//   - 0b10: reject
+	APDispositionField      = binding.NewRegField(0, 21, 22, "APDisposition")
+	DispositionAllowRegMark = binding.NewRegMark(APDispositionField, DispositionAllow)
+	DispositionDropRegMark  = binding.NewRegMark(APDispositionField, DispositionDrop)
+	DispositionRejRegMark   = binding.NewRegMark(APDispositionField, DispositionRej)
+	// reg0[24..26]: Field to indicate the reasons of sending packet to the controller.
+	// Marks in this field include,
```

Comment:
Not only include these marks, but also combinations of these marks, like `0b011` means logging+reject.

Reply:
> Actually, this field includes not only these marks, but also combinations of these marks, like `0b011` means logging+reject.

@GraysonWu Thanks for reminding it. As I didn't find direct match/set for these marks in current code, I would left them to be added when code needs them. What do you think?

Reply:
Works for me.